### ISSUE TITLE
"Connection is secure" is displayed on badssl.com

### ISSUE THREAD

**Issue by SimonBasca** [mozilla-mobile/firefox-ios#9082]
## Steps to reproduce
1. Go to "badssl.com"
2. Tap on "expired"
3. Tap "advance"
### Expected behavior
- "Connection is not secure" message should be presented via the Lock icon
### Actual behavior
- "Connection is secure" message should be presented via the Lock icon
### Device & build information
* Device: iPhone 12 Pro (14.7)
* Build version: 37. 0(5971)
### Notes
Attachments: 
![Screenshot](https://user-images.githubusercontent.com/32130829/132540628-1538d86d-e531-4700-97e4-22372abbc264.jpeg)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-3137)

**Comment by dnarcese** [comment]
@SimonBasca this is expected for now as the feature only checks https status, not ssl.

**Comment by st3fan** [comment]
@dnarcese I am not sure if this should ship if it ignores SSL failures. That is rather serious.

I just tested this too as we want to move similar functionality in Focus and I noticed that it also says "Secure Connection" for Expired, Revoked and Untrusted certificates. In all those cases Desktop will show a little warning sign on top of the lock icon and say "Connection not secure".

I think it is better to not ship this if we cannot properly show the status.
